bin/XMLToSARIF.sh

```diff
@@ -0,0 +1,125 @@
+print_usage() {
+    echo "Usage: XMLToSARIF.sh --inputXmlReport  <inputXmlReport> [--outputSarifReport <outputSarifReport>] [--javaOrToolHomeDir <javaOrToolHomeDir>]"
+    echo ""
+    echo "Options:"
+    echo "  --inputXmlReport      (required) Path to the input XML report."
+    echo "  --outputSarifReport   Path to the output SARIF report."
+    echo "  --javaOrToolHomeDir         Path to the tool home directory."
+    echo ""
+}
+
+check_param() {
+    # check if the parameter is empty or starts with "--"
+    local param_name="$1"
+    local param_value="$2"
+    if [[ -z "$param_value" || "$param_value" == --* ]]; then
+        echo "Error: Empty value passed to option $param_name "
+        exit 1
+    fi
+}
+
+getJavaPath() {
+    local javaOrParasoftToolRootPath="$1"
+
+    if [ ! -d "$javaOrParasoftToolRootPath" ]; then
+        echo "$javaOrParasoftToolRootPath is not found."
+        exit 1;
+    fi
+
+    local javaFileName="java"
+    # bin -- Find java executable in Java home directory
+    # bin/dottest/Jre_x64/bin -- Find java executable in DotTest installation directory
+    # bin/jre/bin -- Find java executable in Jtest,C++Test installation directory
+    local javaPaths=("bin" "bin/dottest/Jre_x64/bin" "bin/jre/bin")
+
+    for path in "${javaPaths[@]}"; do
+        local javaFilePath="$javaOrParasoftToolRootPath/$path/$javaFileName"
+        if [ -f "$javaFilePath" ]; then
+            echo "$javaFilePath"
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+# 1.Initialize variables:
+# Base directory of the transformer
+BASE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+SARIF_XSL_PATH="$BASE_DIR/xsl/sarif.xsl"
+
+java_or_parasoft_tool_root_path=""
+xml_report_path=""
+sarif_report_path=""
+
+# 2. Process options into variables.
+while [[ $# -gt 0 ]] ; do
+  case "$1" in
+    --inputXmlReport)
```

Comment:
It seems that we can easily support short options with the condition `-i|--inputXmlReport)`.
If both a short option and a long option are specified, we just take the last value of the same options, similar to the case if we accidentally specify two long options.